nautobot/core/api/serializers.py

```diff
@@ -889,6 +889,37 @@ class RenderJinjaSerializer(serializers.Serializer):  # pylint: disable=abstract
     """Serializer for RenderJinjaView."""
 
     template_code = serializers.CharField(required=True)
-    context = serializers.DictField(default=dict)
+
+    # JSON context (required for JSON mode, mutually exclusive with object fields)
+    context = serializers.DictField(required=False)
+
+    # Object selection fields (both required for object mode, mutually exclusive with context)
+    content_type = ContentTypeField(
+        queryset=ContentType.objects.all().order_by("app_label", "model"),
+        required=False,
+        allow_null=True,
+    )
+    object_uuid = serializers.UUIDField(required=False, allow_null=True)
+
+    # Read-only response fields
     rendered_template = serializers.CharField(read_only=True)
     rendered_template_lines = serializers.ListField(read_only=True, child=serializers.CharField())
+
+    def validate(self, attrs):
+        """Ensure either context OR object fields are provided, but not both."""
+        has_context = "context" in attrs  # Check presence, not truthiness (allows empty {})
+
+        # Check for meaningful object fields (not just presence)
+        content_type = attrs.get("content_type")
+        object_uuid = attrs.get("object_uuid")
+        has_object = bool(content_type and object_uuid)
```

Comment:
Should we validate that the specified content_type + object_uuid actually exists here? I see you're doing that in the view, just wonder if it makes more sense in the form.